src/Microsoft.Extensions.Logging.EventSource/LoggingEventSource.cs

```diff
@@ -0,0 +1,236 @@
+    /// 
+    /// class MyEventListener : EventListener {
+    ///     protected override void OnEventSourceCreated(EventSource eventSource) {
+    ///         if (eventSource.Name == "Microsoft-Extensions-Logging") {
+    ///             // initialize a string, string dictionary of arguments to pass to the EventSource.  
+    ///             // Turn on loggers matching App* to Information, everything else (*) is the default level (which is EventLevel.Error)
+    ///             var args = new Dictionary&lt;string, string&gt;() { { "FilterSpecs", "App*:Information;*" } };
+    ///             // Set the default level (verbosity) to Error, and only ask for the formatted messages in this case.  
+    ///             EnableEvents(eventSource, EventLevel.Error, LoggingEventSource.Keywords.FormattedMessage, args);
+    ///         }
+    ///     }
+    ///     protected override void OnEventWritten(EventWrittenEventArgs eventData) {
+    ///         // Look for the formatted message event, which has the following argument layout (as defined in the LoggingEventSource.  
+    ///         // FormattedMessage(LogLevel Level, int FactoryID, string LoggerName, string EventId, string FormattedMessage);
+    ///         if (eventData.EventName == "FormattedMessage")
+    ///             Console.WriteLine("Logger {0}: {1}", eventData.Payload[2], eventData.Payload[4]);
+    ///     }
+    /// }
+    /// </summary>
+    [EventSource(Name = "Microsoft-Extensions-Logging")]
+    public class LoggingEventSource : System.Diagnostics.Tracing.EventSource
+    {
+        /// <summary>
+        /// This is public from an EventSource consumer point of view, but since these defintions 
+        /// are not needed outside this class 
+        /// </summary>
+        public class Keywords
+        {
+            /// <summary>
+            /// Meta events are evnets about the LoggingEventSource itself (that is they did not come from ILogger
+            /// </summary>
+            public const EventKeywords Meta = (EventKeywords)1;
+            /// <summary>
+            /// Turns on the 'Message' event when ILogger.Log() is called.   It gives the information in a programatic (not formatted) way  
+            /// </summary>
+            public const EventKeywords Message = (EventKeywords)2;
+            /// <summary>
+            /// Turns on the 'FormatMessage' event when ILogger.Log() is called.  It gives the formatted string version of the information.  
+            /// </summary>
+            public const EventKeywords FormattedMessage = (EventKeywords)4;
+            /// <summary>
+            /// Turns on the 'MessageJson' event when ILogger.Log() is called.   It gives  JSON representation of the Arguments. 
+            /// </summary>
+            public const EventKeywords JsonMessage = (EventKeywords)8;
+        }
+
+        /// <summary>
+        ///  The one and only instance of the LoggingEventSource.  
```

Comment:
That is how EventSource works; there can be only one EventSource instance with a given name per app domain.